app/utils.py

```diff
@@ -47,3 +47,38 @@ def get_contract_by_id(contract_id):
     sf = Salesforce(username=SF_USERNAME, password=SF_PASSWORD, security_token=SF_SECURITY_TOKEN, domain=SF_DOMAIN)
     contract = sf.Contract.get(contract_id)
     return contract
+
+
+def fetch_cases_by_date(case_date_from, case_date_to):
+
+    sf = Salesforce(username=SF_USERNAME, password=SF_PASSWORD, security_token=SF_SECURITY_TOKEN, domain=SF_DOMAIN)
+
+    cases = sf.query(
+        "SELECT id, Contract__c, Description, CaseNumber, Case_URL__c from Case WHERE (Subject LIKE '{r}' \
+         OR Subject like '{n}') AND Contract__c != null"
+        .format(r='RECOUPABLE%', n='NON-RECOUPABLE%'))['records']
+
+    contracts_ids_query = "SELECT Contract__c from Case WHERE (Subject LIKE '{r}' OR Subject like '{n}') \
+    AND Contract__c != null".format(r='RECOUPABLE%', n='NON-RECOUPABLE%')
+
+    contracts = sf.query(
+        "SELECT Id, Eventbrite_Username__c, Hoopla_Account_Name__c, ActivatedDate from Contract WHERE id IN ({q}) \
+        AND BillingCountry = 'Brazil' AND ActivatedDate > {f} AND ActivatedDate < {t}"
+        .format(q=contracts_ids_query, f=case_date_from, t=case_date_to))['records']
+
+    result = []
+
+    for case in cases:
+        for contract in contracts:
```

Comment:
Hacer for loops anidados nunca es la mejor alternativa, fijate si es posible hacer la query mas inteligente para que ya traiga los contratos asociados a un caso con lo que solo deberias recorrer una sola lista